sympy/matrices/matrices.py

```diff
@@ -2360,25 +2395,23 @@ def diagonal_solve(self, rhs):
         return self._diagonal_solve(rhs)
 
     def dot(self, b):
-        """Return the dot product of Matrix self and b relaxing the condition
-        of compatible dimensions: if either the number of rows or columns are
-        the same as the length of b then the dot product is returned. If self
-        is a row or column vector, a scalar is returned. Otherwise, a list
-        of results is returned (and in that case the number of columns in self
-        must match the length of b).
+        """Return the dot product of Matrix self and b only if they are either
```

Comment:
How about

```
Return the dot product of two vectors of equal length. ``self`` must be a ``Matrix`` of size 1 x n or n x 1,
and ``b`` must be either a matrix of size 1 x n, n x 1, or a list of length n. A scalar is returned.
```

Reply:
yup Its better here I think `list/tuple` will be better.